feat #6 : 리뷰 페이지 구현

Gyuuuuuuu opened the pull request:
## 📌 Summary
- Resolve: #6  
- 리뷰 페이지 기능 전체 구현 (리스트, 카드, 작성/수정/삭제, 상세 모달 포함)

<br>

## ✨ Description
- `ReviewCard` 컴포넌트 구현  
- `ReviewList` 페이지에서 리뷰 카드 리스트 출력  
- `ReviewWrite` 컴포넌트로 리뷰 작성 기능 구현  
- `ReviewModify` 컴포넌트로 수정 기능 추가  
- 리뷰 삭제 기능 구현  
- 리뷰 카드 클릭 시 `ReviewDetailModal`로 상세 정보 표시  
- 디렉토리 구조 정리: `Review` → `ReviewPage`로 이동  

<br>

## 📸 Screenshot

| 기능               | 스크린샷/GIF |
|--------------------|--------------|
| 리뷰 리스트 페이지 전체 | ![image](https://github.com/user-attachments/assets/6d36721a-e66d-414b-a240-bf52b09dde8b) |
| 리뷰 작성             | ![write](https://github.com/user-attachments/assets/a94841da-e0af-4f44-8169-bf85acec6da6) |
| 리뷰 상세 모달         | ![Animation](https://github.com/user-attachments/assets/adf53b8c-36f2-483c-9b6b-e85e591a04cf) |
| 리뷰 수정             | ![modify](https://github.com/user-attachments/assets/08ac35d6-bd63-4bab-8395-7395ac699d1e) |
| 리뷰 삭제             | ![delete](https://github.com/user-attachments/assets/42f80ccd-ba18-4899-a036-502251be544e) |

<br>

## 🗒️ Review Point
- 프론트가 익숙치 않아 많이 미흡합니다! 코드 컨벤션에 맞지 않거나 코드가 미흡한 부분에 대해서 지적해주시면 감사하겠습니다.

Gyuuuuuuu committed:
리뷰 페이지, 수정, 삭제, 리뷰 상제 모달

yoohyebin commented on travel-project/src/components/ReviewCard.js:
```diff
@@ -0,0 +1,79 @@
+import './ReviewCard.css';
+
+const BigStar = () => (
+  <svg className="star-svg" width="25" height="25" viewBox="0 0 24 24" style={{display:'inline-block', verticalAlign:'middle'}}>
```
이 부분 inline으로 style 지정하신 이유가 있으실까요?

Gyuuuuuuu replied:
다른 이유는 없고 레퍼런스 삼았던 코드가 인라인으로 작성된 코드였습니다! 가독성 살려서 따로 css 파일로 관리하도록 하겠습니다

yoohyebin reviewed (commented):
star 부분이 반복적으로 등장하는것 같은데 중복을 최소화 하는 방향으로 수정되면 좋을것 같습니다! 어떻게 생각하시나요?

yoohyebin reviewed (commented):
pr 템플릿 맞춰서 pr 내용 수정 및 pr name 수정해주세요

junhyung0806 commented on travel-project/src/pages/Review/ReviewList.js:
```diff
@@ -0,0 +1,60 @@
+function ReviewList({ reviews }) {
+  const navigate = useNavigate();
+
+  
```
공백

Gyuuuuuuu replied:
수정했습니다!

Gyuuuuuuu committed:
style #6 : 전체 폰트 Noto Sans로 통일

Gyuuuuuuu committed:
chore #6 : Review 폴더명을 ReviewPage로 변경

Gyuuuuuuu committed:
style #6 : inline 스타일 제거 및 css 분리

Gyuuuuuuu committed:
style #6 : 수정하기, 삭제하기 버튼 크기 통일

Gyuuuuuuu committed:
refactor #6 : review 페이지 별점 기능 컴포넌트화

Gyuuuuuuu committed:
refactor #6 : ReviewCard에 StarRating 컴포넌트 사용

Gyuuuuuuu committed:
style #6 : ReviewCard.css 수정, card 사진 여백 제거

Gyuuuuuuu committed:
chore #6 : 파일 확장자 .jsx로 통일

Gyuuuuuuu committed:
style #6 : 상단바 css 폰트 크기 변경

Gyuuuuuuu committed:
refactor #6 : ReviewDetail파일 StarRating 컴포넌트 사용으로 변경

Gyuuuuuuu committed:
chore #6 : 샘플 데이터 삭제

alsshsskssl commented on travel-project/src/App.js:
```diff
@@ -1,24 +1,46 @@
+
 import './App.css';
 
 function App() {
```
App()에 바로 리뷰 페이지 작성보다 엔드포인트만 정의해주시면 됩니다. 헤더랑 푸터 사이에 엔드포인트들 끼어 넣으시면 될듯합니다

Gyuuuuuuu committed:
refactor #6 : App.js 파일에 사용되었던 함수들 제거 및 ReviewContext파일 생성

Gyuuuuuuu committed:
docs #6 : StarTating 주석 추가

Gyuuuuuuu committed:
refactor #6 : ReviewContext로 state받기

Gyuuuuuuu committed:
docs #6 : app.js 주석추가

Gyuuuuuuu committed:
style #6 : region카드와 review card 디자인 통일

Gyuuuuuuu committed:
style #6 : review detail modal 스타일 수정

- modal창 전체 크기 확대
- modal 리뷰 내용 컨테이너 생성으로 관리
- modal창 X 아이콘 위치 변경
- modal창 수정 및 삭제하기 버튼 위치 변경

Gyuuuuuuu committed:
style #6 : 리뷰페이지 한 줄당 card 5개씩 나오도록 수정

Gyuuuuuuu committed:
file #6 : reviewPage 전용 json과 이미지 파일 추가

Gyuuuuuuu committed:
feat #6 : 리뷰 작성 및 수정 페이지에서 여행 시작 날짜, 끝 날짜 선택 추가

Gyuuuuuuu committed:
refactor #6 : review page에서 사용할 사진 및 json 데이터 경로 수정

Gyuuuuuuu committed:
Merge branch 'develop' into feat/#6
